crates/revm/src/instructions.rs

```diff
@@ -74,175 +72,165 @@ pub enum Return {
-        opcode::PUSH27 => stack::push::<27>(interp),
-        opcode::PUSH28 => stack::push::<28>(interp),
-        opcode::PUSH29 => stack::push::<29>(interp),
-        opcode::PUSH30 => stack::push::<30>(interp),
-        opcode::PUSH31 => stack::push::<31>(interp),
-        opcode::PUSH32 => stack::push::<32>(interp),
-        opcode::DUP1 => stack::dup::<1>(interp),
-        opcode::DUP2 => stack::dup::<2>(interp),
-        opcode::DUP3 => stack::dup::<3>(interp),
-        opcode::DUP4 => stack::dup::<4>(interp),
-        opcode::DUP5 => stack::dup::<5>(interp),
-        opcode::DUP6 => stack::dup::<6>(interp),
-        opcode::DUP7 => stack::dup::<7>(interp),
-        opcode::DUP8 => stack::dup::<8>(interp),
-        opcode::DUP9 => stack::dup::<9>(interp),
-        opcode::DUP10 => stack::dup::<10>(interp),
-        opcode::DUP11 => stack::dup::<11>(interp),
-        opcode::DUP12 => stack::dup::<12>(interp),
-        opcode::DUP13 => stack::dup::<13>(interp),
-        opcode::DUP14 => stack::dup::<14>(interp),
-        opcode::DUP15 => stack::dup::<15>(interp),
-        opcode::DUP16 => stack::dup::<16>(interp),
+        opcode::STOP => return_stop(interp, host),
+        opcode::ADD => arithmetic::wrapped_add(interp, host),
+        opcode::MUL => arithmetic::wrapping_mul(interp, host),
+        opcode::SUB => arithmetic::wrapping_sub(interp, host),
+        opcode::DIV => arithmetic::div(interp, host),
+        opcode::SDIV => arithmetic::sdiv(interp, host),
+        opcode::MOD => arithmetic::rem(interp, host),
+        opcode::SMOD => arithmetic::smod(interp, host),
+        opcode::ADDMOD => arithmetic::addmod(interp, host),
+        opcode::MULMOD => arithmetic::mulmod(interp, host),
+        opcode::EXP => arithmetic::eval_exp::<S>(interp, host),
+        opcode::SIGNEXTEND => arithmetic::signextend(interp, host),
+        opcode::LT => bitwise::lt(interp, host),
+        opcode::GT => bitwise::gt(interp, host),
+        opcode::SLT => bitwise::slt(interp, host),
+        opcode::SGT => bitwise::sgt(interp, host),
+        opcode::EQ => bitwise::eq(interp, host),
+        opcode::ISZERO => bitwise::iszero(interp, host),
+        opcode::AND => bitwise::bitand(interp, host),
+        opcode::OR => bitwise::bitor(interp, host),
+        opcode::XOR => bitwise::bitxor(interp, host),
+        opcode::NOT => bitwise::not(interp, host),
+        opcode::BYTE => bitwise::byte(interp, host),
+        opcode::SHL => bitwise::shl::<S>(interp, host),
+        opcode::SHR => bitwise::shr::<S>(interp, host),
+        opcode::SAR => bitwise::sar::<S>(interp, host),
+        opcode::SHA3 => system::sha3(interp, host),
+        opcode::ADDRESS => system::address(interp, host),
+        opcode::BALANCE => host::balance::<S>(interp, host),
+        opcode::SELFBALANCE => host::selfbalance::<S>(interp, host),
+        opcode::CODESIZE => system::codesize(interp, host),
+        opcode::CODECOPY => system::codecopy(interp, host),
```

Comment:
Do you for sure know that the specific change is what introduced the speed improvement? What about making the PR only change the benchmarks first, and then check the is_static change in one PR then this function sig change?

Reply:
Yes, i tested it while working on it. And this is second time I did this change. Similar results.